File: Java/1309. Decrypt String from Alphabet to Integer Mapping.java
class Solution {
    public String freqAlphabets(String s) {
        int i = 0, a = 'a', l = s.length();
        StringBuilder sb = new StringBuilder();
        
        while (i < l) {
            if (i < l - 2 && s.charAt(i+2) == '#') {
                sb.append((char) ((int)Integer.valueOf(s.substring(i, i+2)) + a - 1));
                i += 2;
            } else {
                sb.append((char) (Character.getNumericValue(s.charAt(i)) + a - 1));
            }
            i++;
        }
        
         return sb.toString();
    }
}

File: Python3/1309. Decrypt String from Alphabet to Integer Mapping.py
class Solution:
    def freqAlphabets(self, s: str) -> str:
        a = ord('a')
        ans = ''
        i, L = 0, len(s)
        
        while i < L:
            #print(s[i])
            # this is an over-9 char
            if i < L-2 and s[i+2] == '#':
                ans += chr(int(s[i:i+2]) + a - 1)
                i += 2
            else:
                ans += chr(int(s[i]) + a - 1)
            i += 1
        
        return ans